### PR TITLE
Chart

### LINKED ISSUE #45: Graphs - Admin page

### PR DESCRIPTION
**Pull request by bertacsevercan**
#45 
- Add statistical information such as; daily. monthly, yearly, and total coupon sales. These fields are not certain though we can change them later.
- Add the bar graph for different coupon types sales, showing them over months.
- All of the data is static, coming from firebase.
- I added this to AdminDashboard container, but it will create conflicts. When Admin dashboard merges, I will solve those conflicts for this component.
- Desktop
Ekran Resmi 2020-11-17 16 47 55

- iPad pro

Ekran Resmi 2020-11-17 16 52 41

- iPad

Ekran Resmi 2020-11-17 16 52 34

- mobile

Ekran Resmi 2020-11-17 16 52 24

### CODE REVIEW

File: src/components/CouponsChart/index.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  useEffect(() => {
+    const unsubscribe = db
+      .collection("soldCoupons")
+      .where("compName", "==", "Arby's")
+      .onSnapshot((snapshot) => {
+        const dataArr = [];
+        snapshot.forEach((doc) => {
+          dataArr.push({ ...doc.data(), docId: doc.id });
+        });
+        setSoldCoupons(dataArr[0]);
```
Comment (halitbatur) [inline src/components/CouponsChart/index.jsx:25]: u are getting all the sold coupons and then only using the first one to set the state why is that ?

Comment (bertacsevercan) [reply src/components/CouponsChart/index.jsx:25]: No, I'm not getting all of them I have a where clause and only getting one of them. So, there is only one doc that I'm using which is inside dataArr

**Comment by bertacsevercan** [comment]
- New grid layout to statistics.
- Mobile
Ekran Resmi 2020-11-18 14 40 50

- İpad
Ekran Resmi 2020-11-18 14 41 00

- iPad pro
Ekran Resmi 2020-11-18 14 41 12

File: src/components/CouponsChart/index.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  return (
+    <div className="couponsChart">
+      <Divider className="couponsChartTitle">
+        <Title level={3}>{t("couponsChart.title")}</Title>
+      </Divider>
+      {loading ? (
+        <Spin className="spin" size="large" tip="Loading..." />
+      ) : (
```
Comment (ihab-sensei) [inline src/components/CouponsChart/index.jsx]: I would use a `t` hook here in `tip`

**Review by ihab-sensei — CHANGES_REQUESTED** [review]

File: src/components/CouponsChart/index.jsx (hunk 2 above)
Comment (bertacsevercan) [reply src/components/CouponsChart/index.jsx]: Oh yeah I kind of forgot about that!

**Review by ihab-sensei — APPROVED** [review]
LGTM!

**Review by halitbatur — APPROVED** [review]

**Pull request merged**